frontend/src/App.vue

```diff
@@ -166,13 +178,32 @@ export default {
       browserLastRoute: {},
       viewerLastRoute: {},
       isMobileMenu: false,
+      showGemSearch: false,
     };
   },
   computed: {
     ...mapState({
       model: state => state.models.model,
     }),
   },
+  created() {
+    // add handler for escape key
+    window.addEventListener('keydown', (e) => {
+      if (e.keyCode === 27 && this.showGemSearch) {
+        e.preventDefault();
+        this.showGemSearch = false;
+      }
+    });
+  },
+  methods: {
+    handleBodyClick(e) {
+      const navbar = document.querySelector('#navbar');
+
+      if (!navbar.contains(e.target) && this.showGemSearch) {
+        this.showGemSearch = false;
+      }
+    },
```

Comment:
This function is essentially running on every click on the website, which we should try to avoid if possible.
The previous `GemSearch` used a different implementation
```
@focus="showResults = true"
@blur="blur()">
```
Could a similar implementation be used here?

Reply:
That's a bit harder to do now, for example there is a select now as well. Clicking on the HTML body except something is a common pattern though. Is there any specific scenarios where this would cause problems?

Reply:
As a programming principle, I think it is better to apply changes as locally as possible. Overloading all clicks on the website feels unnecessary, and it may have side-effects (although maybe not at the moment). I've figured out an alternative approach, have a look [here](https://github.com/MetabolicAtlas/MetabolicAtlas/compare/feat/menu-bar-search...feat/blur-menu-bar-search).

Reply:
Although I agree with the principle in general this is still a common exception. We are not overriding any behaviors, it's rather that all clicks which do not fall into this if condition simply behave as they usually would.

I tried your new branch and it doesn't work if you click outside the search area right after the search shows. Currently you need to click either the select or the input first, and then click outside.

Reply:
That behaviour is because of the broken `focus()` I mentioned below.

Reply:
With the auto-focus working I see it works to hide it directly now with both clicking outside and ESC so that's good. But I notice that I can't click on any search results now, looks like maybe the blur is happening before the click.

Also the previous behavior was clicking outside the navbar hides it the search but the new behavior is clicking outside the search component. So even if someone clicks inside the navbar but outside the search it still hides the search. This is maybe an ok change but I feel like the previous one was more straightforward.

Reply:
The blur was indeed happening too soon. In the previous implementation of the GemSearch it was 200ms, now it's 100 and it seems to work fine.

Regarding the second point, I think the new implementation is in line with the inspiration from [Apple's website](https://apple.com).

Reply:
The blur seems a bit hit-or-miss for me now. Result still not going through sometimes, is it working for you 100%?

> Regarding the second point, I think the new implementation is in line with the inspiration from Apple's website.

Ok I can buy that. :)

Reply:
I've now increased the delay until blurring occurs to 200ms, I hope it's okay. For me 100ms was enough, but you're right it's better to be on the safe side.